packages/react-components/src/echarts/extensions/minMaxDataStreamValues/updateDataStreamMinMax.ts

```diff
@@ -0,0 +1,54 @@
+import { EChartsExtensionInstallRegisters } from 'echarts/types/src/extension';
+import { findMinMax } from './minMaxDataStreamSync';
+import useDataStore from '../../../store';
+import throttle from 'lodash.throttle';
+import Grid from 'echarts/types/src/coord/cartesian/Grid';
+
+const THROTTLE_RATE = 2000;
+
+// Echarts core use type does not map correctly to the echarts extension type so exporting as any
+// eslint-disable-next-line
+export const dataStreamMinMaxSyncExtension: any = throttle((registers: EChartsExtensionInstallRegisters) => {
+    registers.registerUpdateLifecycle(
+      'series:afterupdate',
+      (_ecModel, _api, params) => {
+        const appKitChartId = _ecModel.option.appKitChartId as string; // widget-id
+        const grid = _api
+          ?.getCoordinateSystems()
+          .find((a) => a.model?.type === 'grid') as Grid;
+        const axis = grid?.getAxis('x');
+        const start = axis?.coordToData(axis.toLocalCoord(0));
+        const end = axis?.coordToData(axis.toLocalCoord(axis?.getExtent()[1]));
+
+        if (
+          !params.updatedSeries ||
+          (!start && start !== 0) ||
```

Comment:
what is this guard for? It doesn't make sense for start or end to be 0, but why do we need to check against it?